appengine/static/js/rosebotics.js

```diff
@@ -3,34 +3,20 @@ var loginModalFunctions = function() {
 	var modal = $("#edit-student-details-modal");
 	var student_name_field = $('#edit-student-details-modal-name-field');
   var rosefire_button = $('input[name="getToken"]');
-  rosefire_button.click(function() {
-    $('#rosefire-modal').modal('show', {backdrop: false});
-    $('#rosefire-error').addClass('hidden');
-  });
-  var rosefire_login = $('#rosefire-login');
   var logging_in = false;
-  rosefire_login.click(function() {
-    var rose_email = $('#rose-email').val();
-    var rose_password = $('#rose-password').val();    
+  rosefire_button.click(function() {
     if (!logging_in) {
       logging_in = true;
-      var creds = {
-        email: rose_email + "@rose-hulman.edu",
-        password: rose_password,
-        registryToken: '85bee1dadcaf0093246a8b3b6c627545a887539d92f71480ec70ba34c67a5f2dc736c47e0c04039844eef9dd9f2ea7b8kENtxsdDt+Oo9mGExHXYBR56vm1r31eCnF0OVXIPQbY1ExYlnCQMZteHQIuCZ4AS/BrFZs9GWNAbit+ph74f5geBc6lN4ZaS3NomanSmFiM='
-      };
-      Rosefire.getToken(creds, function(err, token) {
-        if (err) {
-          $('#rosefire-error').removeClass('hidden');
-        } else {
-          $('#rosefire-error').addClass('hidden');
+      var registryToken = 'ef9309a0f6789c5e9ae471609c2e0372a75a5edfdf8b57cd6a758bf49bfae9471f96b4a56b0c119bd04a2168958cb97eFjjRfqJBcZKe7B/AshEyeybTgYJqfrAWycL++Q6/e74YWX9ekdDCEeJPIxZ3HL37VROZPuNs9ys/xuYK389NjQ==';
+      Rosefire.signIn(registryToken, function(err, token) {
+        if (!err) {
+          var uid = JSON.parse(atob(token.split('.')[1])).d.uid;
```

Comment:
This just decodes the JWT. Ideally this logic could be simplified within my library, but I haven't gotten there yet 😃